python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -314,6 +315,79 @@ def getRun(self, collection=None, id=None):
                           pipeline=result['pipeline_id'])
         return run
 
+    def addStorageInfo(self, ref, storageInfo):
+        """Add storage information for a given dataset.
+
+        Typically used by `Datastore`.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            A reference to the dataset for which to add storage information.
+        storageInfo : `StorageInfo`
+            Storage information about the dataset.
+        """
+        datasetStorageTable = self._schema.metadata.tables['DatasetStorage']
+        with self._engine.begin() as connection:
+            connection.execute(datasetStorageTable.insert().values(dataset_id=ref.id,
+                                                                   datastore_name=storageInfo.datastoreName,
+                                                                   md5=storageInfo.md5,
+                                                                   size=storageInfo.size))
+
+    def updateStorageInfo(self, ref, datastoreName, storageInfo):
+        """Update storage information for a given dataset.
+
+        Typically used by `Datastore`.
+
+        Parameters
+        ----------
+        ref : `DatasetRef`
+            A reference to the dataset for which to add storage information.
+        datastoreName : `str`
+            What datastore association to update.
+        storageInfo : `StorageInfo`
+            Storage information about the dataset.
+        """
+        datasetStorageTable = self._schema.metadata.tables['DatasetStorage']
+        with self._engine.begin() as connection:
+            connection.execute(datasetStorageTable.update().where(and_(
+                datasetStorageTable.c.dataset_id == ref.id,
+                datasetStorageTable.c.datastore_name == datastoreName)).values(
```

Comment:
I'm confused by the relationship between `datastoreName` and `storageInfo.datastoreName`. This may partly be because I'm not sure what a datastore name really is. In this code you update the row that has the supplied `datastoreName` by changing it to be the name stored in the storage info?